homeassistant/components/advantage_air/__init__.py

```diff
@@ -0,0 +1,92 @@
+"""Advantage Air climate integration."""
+
+import collections.abc
+from datetime import timedelta
+import logging
+
+from advantage_air import ApiError, advantage_air
+
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PORT
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import ADVANTAGE_AIR_RETRY, DOMAIN
+
+ADVANTAGE_AIR_SYNC_INTERVAL = 15
+ADVANTAGE_AIR_PLATFORMS = ["climate"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def update(original, updates):
+    """Deep update a dictionary."""
+    for key, val in updates.items():
+        if isinstance(val, collections.abc.Mapping):
+            original[key] = update(original.get(key, {}), val)
+        else:
+            original[key] = val
+    return original
+
+
+async def async_setup(hass, config):
+    """Set up AdvantageAir."""
+    hass.data[DOMAIN] = {}
+    for platform in ADVANTAGE_AIR_PLATFORMS:
+        hass.async_create_task(
+            hass.helpers.discovery.async_load_platform(platform, DOMAIN, {}, config)
+        )
```

Comment:
This is only needed when providing yaml config.

Reply:
Removed. Thanks